qiskit/optimization/problems/base.py

```diff
@@ -35,34 +34,16 @@ def __init__(self):
         """
         self._index = NameIndex()
 
-    def get_indices(self, *name) -> Union[int, List[int]]:
-        """Converts from names to indices.
-
-        If name is a string, get_indices returns the index of the
-        object with that name.  If no such object exists, an
-        exception is raised.
-
-        If name is a sequence of strings, get_indices returns a list
-        of the indices corresponding to the strings in name.
-        Equivalent to map(self.get_indices, name).
-
-        See `NameIndex.convert` for details.
-
-        If the subclass does not provide an index function (i.e., the
-        interface is not indexed), then a NotImplementedError is raised.
+    def get_index(self, name: Union[str, int]) -> int:
+        """Converts from a name to index.
+        """
+        return self._index.convert(name)
 
-        Example usage:
+    def get_indices(self, names: List[Union[str, int]]) -> List[int]:
```

Comment:
Should we merge `get_index` and `get_indices` by allowing `get_index` to take a list, too? I imagine this would be handy to not have to distinguish between the two as a user.

Reply:
`BaseInterface` will be deleted to simplify the class organization. No class uses `get_indices` and `get_index`.

Reply:
Great, even better!